crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -995,6 +995,10 @@ where
             }
             Err(err) => {
                 debug!(target: "consensus::engine", ?err, "Failed to insert downloaded block");
+                if !matches!(err.kind(), InsertBlockErrorKind::Internal(_)) {
+                    // non-internal error kinds occurr if the payload is invalid
+                    self.invalid_headers.insert(block.header);
```

Comment:
the error contains the full invalid block, so we don't need to clone above,

we can add a helper here that consumes the type and returns the block `.into_block`

https://github.com/paradigmxyz/reth/blob/3bd26b04410047161591bb6dd933cc272b2a6a84/crates/interfaces/src/blockchain_tree/error.rs#L74-L78

Reply:
nice, not sure how I missed that the error contained the block! added this